### ISSUE TITLE
Contour Labelling bug

### ISSUE THREAD

**Issue by mikewalker92** [SciTools/cartopy#327]
```
import iris
import cartopy
import matplotlib.pyplot as plt
import iris.quickplot as qplt
import cartopy.crs as ccrs


cube = iris.load_cube(iris.sample_data_path('A1B.2098.pp'))

ax = plt.axes(projection=ccrs.Orthographic())

plt.gca().set_global()

im = qplt.contour(cube)
plt.clabel(im, inline=1, fontsize=8)


plt.show()

```

this code produces the following error message;

```
Traceback (most recent call last):                                                                                                                 
  File "bugLabels.py", line 16, in <module>                                                                                                        
    plt.clabel(im, inline=1, fontsize=8)                                                                                                           
  File "/usr/local/sci/lib/python2.7/site-packages/matplotlib/pyplot.py", line 2467, in clabel                                                     
    ret = ax.clabel(CS, *args, **kwargs)
  File "/usr/local/sci/lib/python2.7/site-packages/matplotlib/axes.py", line 7730, in clabel
    return CS.clabel(*args, **kwargs)
  File "/usr/local/sci/lib/python2.7/site-packages/matplotlib/contour.py", line 220, in clabel
    self.labels(inline,inline_spacing)
  File "/usr/local/sci/lib/python2.7/site-packages/matplotlib/contour.py", line 651, in labels
    inline_spacing )
  File "/usr/local/sci/lib/python2.7/site-packages/matplotlib/contour.py", line 431, in calc_label_rot_and_inline
    extrap=True )
  File "/usr/local/sci/lib/python2.7/site-packages/matplotlib/mlab.py", line 2840, in less_simple_linear_interpolation
    jj = max(jj)
ValueError: max() arg is an empty sequence
```

I thought initially that this was probably linked to the same scaling issues scaling issues (seen before in #321) causing the plot to be so small that the correct arguments were not getting passed onto matplotlib. However if this is the casue, then it is weird that the set_global command would not then fix the problem no matter where it was placed, and it seemed strange that if the file was changed to air_temp.pp, then it would plot without issue, despite the fact that the plots without labels and without set_gobal look identical.

**Comment by akrherz** [comment]
For whatever it is worth, I attempted to reproduce this bug with a current (except pandas) conda-forge stack on python2.7/linux64 and could not.  The example script produces this plot
![figure_1](https://user-images.githubusercontent.com/210858/33382183-a6cb58f0-d4e5-11e7-9b55-ccbfb3299d69.png)

**Comment by pelson** [comment]
Contour labels covered by #13, which should be fixed by #1257.
